roles/chameleon_mariadb/templates/mariadb.j2

```diff
@@ -45,14 +45,22 @@ mysqldump --defaults-extra-file=$TMPFILE --opt --flush-logs --single-transaction
  ${ADDITIONAL_OPTIONS} \
  --all-databases | bzcat -zc > ${DIR}/${PREFIX}`date +%Y%m%d-%H%M%S`.sql.bz2
 
-if [ $? -eq 0 ] ; then
-    cleanup
-    touch /tmp/mysqlbackup_success
-fi
+# AWS S3  cross site backup
+# Copy latest backup to /etc/awsconf to mount to container
+S3DIR="{{ mariadb_backup_s3_conf_dir }}"
+LATEST_FILE=`ls -rt ${DIR} | grep mysql_backup | tail -1`
+cp $DIR/$LATEST_FILE $S3DIR
 
-rsync -vrlptz ${DIR} "{{ mariadb_backup_destination_host }}":"{{ mariadb_backup_destination_dir }}" 2>&1 | /usr/bin/logger -t "{{ mariadb_backup_log_key }}" && ssh "{{ mariadb_backup_destination_host }}" 'find {{ mariadb_backup_destination_dir }}* -maxdepth 2 -mtime "{{ mariadb_backup_retention }}" -exec rm {} \;'
+# Create bucket and copy backup file, cleanup file
+docker run -e AWS_ACCESS_KEY_ID="{{ mariadb_backup_s3_key_id }}" -e AWS_SECRET_ACCESS_KEY="{{ mariadb_backup_s3_key }}" --rm -it -v ${S3DIR}:/root/.aws amazon/aws-cli --endpoint="{{ mariadb_backup_s3_endpoint }}" --region=us-east-1 s3 mb s3://"{{ mariadb_backup_container_name }}"
```

Comment:
Does `--region` need to be us-east-1? Does it just need to be set to a value? If so, maybe set it to `dummy` or something so it's clear it's not really being used.

Reply:
I thought it did....but apparently it likes Dummy :)

Reply:
So it likes dummy if I just do an 'ls' on the bucket, but if I try to create a bucket it says "The specified location-constraint is not valid", so I'll revert.